FP-Erlang/Week 6/Rahul/assignment5.erl

```diff
@@ -0,0 +1,66 @@
+-module(assignment5).
+-export([add/1, subtract/1, multiply/1, divide/1, greaterThanX/1, lessThanX/1,
+ derive/2, integral/4, foldl/3, map/2, filterl/2, lessthanList/2, sumList/1,
+ squareList/1, evenList/1, factList/1, foldr/3, foldl2/3]).
+
+add(X) -> fun(Y) -> X+Y end.
+
+subtract(X) -> fun(Y) -> X-Y end.
+
+multiply(X) -> fun(Y) -> X*Y end.
+
+divide(X) -> fun(Y) -> X/Y end.
+
+isremdr(X) -> fun(Y) -> Y rem X == 0 end.
+
+greaterThanX(X) -> fun(Y) -> Y>X end.
+
+lessThanX(X) -> fun(Y) -> X>Y end.
```

Comment:
Include the two argument versions of the functions by making implicit calls to the curried functions.